apps/frontend/core-dapp/src/features/simulator/SimulatorSummary.tsx

```diff
@@ -40,53 +40,78 @@ const Nowrap = styled.span`
 // e.g. if enter price is $0.3 and exit at $0.6, position is 200%
 // the subtract part computes the pnl in percentage
 // e.g. with the $0.3 -> $0.6 example, 200% - 100% -> 100% profit
-const getPnlPercentage = (entry?: number, exit?: number): number | undefined => {
+export const getPnlPercentage = (entry?: number, exit?: number): number | undefined => {
   if (entry === undefined || exit === undefined) return undefined
   return (1 / entry) * exit - 1
 }
 
-const SimulatorSummary: React.FC<Props> = ({ payoutRange, entryPrice, exitPrice, tradeSize }) => {
-  const { balancerStore } = useRootStore()
-  const estimatedPnl = getPnlPercentage(entryPrice, exitPrice)
-
+export function getMaxProfitLoss({
+  entryPrice,
+  payoutRange,
+}: {
+  entryPrice: number | undefined
+  payoutRange: Range | undefined
+}): {
+  maxProfit: number | undefined
+  maxLoss: number | undefined
+} {
   const maxProfit = getPnlPercentage(entryPrice, payoutRange?.[1])
   const maxLoss = getPnlPercentage(entryPrice, payoutRange?.[0])
 
-  const renderPnlUI = (
-    pnlPercentage?: number,
-    options: {
-      color?: keyof Color
-      prefix?: string
-    } = {}
-  ): React.ReactNode => {
-    const { color, prefix } = options
-
-    if (pnlPercentage === undefined)
-      return <ColoredLabel color={color ?? 'success'}>???</ColoredLabel>
-
-    const pnlAmountInETHBN =
-      tradeSize === undefined
-        ? undefined
-        : balancerStore.getWstEthAmountInEth(
-            parseEther(`${Math.abs(tradeSize * pnlPercentage).toFixed(18)}`)
-          )
-
-    const pnlAmountInETH =
-      pnlAmountInETHBN === undefined ? undefined : +formatEther(pnlAmountInETHBN)
+  return {
+    maxProfit,
+    maxLoss,
+  }
+}
 
-    const pnlAmount =
-      pnlAmountInETH === undefined || pnlAmountInETH === 0 ? '' : `(${displayEth(pnlAmountInETH)})`
+export const ProfitLossValue: React.FC<{
+  className?: string
+  color?: keyof Color
+  pnlPercentage?: number
+  prefix?: string
+  tradeSize: number | undefined
+}> = observer(({ className, color, pnlPercentage, prefix, tradeSize }) => {
+  const { balancerStore } = useRootStore()
 
+  if (pnlPercentage === undefined)
     return (
-      <p>
-        <ColoredLabel color={color ?? (pnlPercentage >= 0 ? 'success' : 'error')}>
-          {prefix ?? (pnlPercentage >= 0 ? '+' : '-')}
-          {(Math.abs(pnlPercentage) * 100).toFixed(2)}%
-        </ColoredLabel>{' '}
-        <Nowrap>{pnlAmount}</Nowrap>
+      <p className={className}>
+        <ColoredLabel color={color ?? 'success'}>
+          ???
+        </ColoredLabel>
       </p>
```

Comment:
My IDE is complaining that these should be single line, did you get that complain from eslint? 

```suggestion
      <p className={className}>
        <ColoredLabel color={color ?? 'success'}>???</ColoredLabel>
      </p>
```